lib.js

```diff
@@ -7,7 +7,66 @@
  * @param {Filter} filter
  */
 function Iterator(friends, filter) {
-    console.info(friends, filter);
+
+    if (!(filter instanceof Filter)) {
+        throw new TypeError();
+    }
+
+    this.friends = getFriendsToInvite(friends).filter((arg) => filter.isSuitable(arg));
```

Comment:
Можно просто .filte(isSuitable)

Reply:
Так не будет работать, т к теряется контекст

Reply:
?

Я имел в виду, что можно вот так
```suggestion
    this.friends = getFriendsToInvite(friends).filter(filter.isSuitable);
```

Так будет работать.